Fix FVSAND_GPU_KERNEL_LAUNCH macro

gzagaris opened the pull request:
Summary
========

This PR fixes the compilation issues with `FVSAND_GPU_KERNEL_LAUNCH` macro, introduced in 65d704d967db21a1d200569da255eebb08070521. Moreover, changes the `LocalMesh::CreateGridMetrics` routine to use the `FVSAND_GPU_KERNEL_LAUNCH` macro instead.

gzagaris committed:
BUGFIX: fix FVSAND_GPU_KERNEL_LAUNCH macro

The NVCC compiler did not like the constexpr arguments used to
specify SHARED_MEM and CUDA_STREAM leading to compilation errors.
This commit fixed that by specifying literals directly inside th
macro.

gzagaris committed:
ENH: use FVSAND_GPU_KERNEL_LAUNCH instead

Change the code in LocalMesh::CreateGridMetrics to use the
FVSAND_GPU_KERNEL_LAUNCH macro instead.

jsitaraman commented:
Thanks George. On the first call the number of threads N=(ncells+nthreads)*3, since I was trying to do the x, y- and z- coords of cell center all together.  I see that  the argument to the function is still (ncells+nthreads). It's a bit confusing. I will fix that. Otherwise LGTM

gzagaris commented:
> On the first call the number of threads N=(ncells+nthreads)*3, since I was trying to do the x, y- and z- coords of cell center all together. I see that the argument to the function is still (ncells+nthreads). It's a bit confusing. I will fix that. Otherwise LGTM

I see. Thanks for clarifying -- I thought the iteration loop was looping over the interval `[0, ncells+nhalo)`...

gzagaris committed:
BUGFIX: fix cell_center kernel launch

The cell_center kernel loops over N*3 since the kernel
handles all dimensions at once. Fixed that.

gzagaris committed:
ENH: change default block size to 256

gzagaris commented on src/LocalMesh.C:
```diff
@@ -172,26 +172,18 @@ void LocalMesh::CreateGridMetrics()
-  			 center_d,x_d,nvcft_d,cell2node_d,ncells+nhalo);
+
+  const int N = ncells + nhalo;
+  FVSAND_GPU_KERNEL_LAUNCH( cell_center, (N*3), center_d, x_d, nvcft_d, 
```
@jsitaraman -- is this what you'd expect this call to be?

gzagaris commented:
@jsitaraman -- if you are OK with it, can we merge this?
